src/clj_irods/core.clj

```diff
@@ -0,0 +1,205 @@
+    [from-listing listing-extract-fn user zone path]
+    [from-stat stat-extract-fn path]))
+
+(defn object-type
+  [irods user zone path]
+  (let [listing-extract-fn (fn [item] (condp = (:type item) "dataobject" :file "collection" :dir nil))]
+    (from-stat-or-listing :type listing-extract-fn irods user zone path)))
+
+(defn date-modified
+  [irods user zone path]
+  (from-stat-or-listing :date-modified (comp (partial * 1000) #(Integer/parseInt %) :modify_ts) irods user zone path))
+
+(defn date-created
+  [irods user zone path]
+  (from-stat-or-listing :date-created (comp (partial * 1000) #(Integer/parseInt %) :create_ts) irods user zone path))
+
+(defn file-size
+  [irods user zone path]
+  (from-stat-or-listing :file-size :data_size irods user zone path))
+
+(defn uuid
+  [irods user zone path]
+  (cached-or-get irods
+    [from-listing :uuid user zone path]
+    [(fn [cache? irods path]
+       (let [get-from-metadata (fn [metadata]
+                                 (when-let [uuid-meta (first (filter #(= (:attr %) "ipc_UUID") @metadata))]
+                                   (delay (get uuid-meta :value))))]
+           (when-let [metadata (if cache?
+                                 (jargon/cached-get-metadata irods path)
+                                 (jargon/get-metadata irods path :known-type @(object-type irods user zone path)))]
+             (or
+               (get-from-metadata metadata)
+               nil))))
+     path]))
+
+(defn permission
+  [irods user zone path]
+  (cached-or-get irods
+    [from-listing (comp jargon-perms/fmt-perm :access_type_id) user zone path]
+    [(fn [cache? irods user path]
+       (if cache?
+         (jargon/cached-permission-for irods user path)
+         (delay
+           @(jargon/permission-for irods user path :known-type @(object-type irods user zone path)))))
+     user path]))
+
+(defn folder-listing
+  [irods user zone path & {:keys [entity-type sort-column sort-direction limit offset info-types] :or {entity-type :any sort-column :base-name sort-direction :desc limit 1000 offset 0 info-types []}}]
+  (let [cached (get-in
+                 (icat/merge-listings
+                   (icat/filtered-cached-listings irods user zone path
+                                                  :entity-type entity-type
+                                                  :sort-column sort-column
+                                                  :sort-direction sort-direction
+                                                  :info-types info-types))
+                 [entity-type info-types sort-column sort-direction]) ;; don't include limit/offset here
+        cached-range (and cached (icat/get-range cached limit offset))]
+    (if cached-range
+      cached-range
+      (icat/paged-folder-listing irods user zone path :entity-type entity-type :sort-column sort-column :sort-direction sort-direction :limit limit :offset offset :info-types info-types))))
+
+(defn items-in-folder
```

Comment:
This one's a bit simpler, basically, get all cached listings that match entity and info type filters, then grab the first value and pull out the total value field added by my last clj-icat-direct PR. If there's nothing, do a paged folder listing with the defaults and grab it from that. My only concern is really just if the defaults are good.

Neither of these listing-based things has a jargon fallback at this time, though it's probably possible (if horribly slow). Project for later.